### PR TITLE
ghc: allow building with current Sphinx

### PR DESCRIPTION
**Pull request by chexum**
./srcpkg build ghc fails with current tools:

"inplace/bin/ghc-stage1" [...] /build/cbits/longlong.p_o
Running Sphinx v4.0.2

Configuration error:
There is a syntax error in your configuration file: (unicode error) 'unicodeescape' codec can't decode bytes in position 1-2: truncated \uXXXX escape (conf.py, line 87)

...
make: *** [Makefile:128: all] Error 2

According to https://gitlab.haskell.org/ghc/ghc/-/issues/19962

>> This fix is to use raw strings, as already done in master:
>> https://gitlab.haskell.org/ghc/ghc/-/blob/master/docs/users_guide/conf.py#L96

Sanitized from commit (only small parts apply)
https://gitlab.haskell.org/ghc/ghc/-/commit/83407ffc7acc00cc025b9f6ed063add9ab9f9bcc

#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [x] I generally don't use the affected packages but briefly tested this PR

[skip CI] build might take over 3 hours with a single CPU - local build/pkg tested

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (x86_64)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl

### CODE REVIEW

**Comment by chexum** [comment]
By the way, it seems the build checks are failing because there is no check target:

2021-07-10T22:39:46.1203634Z ===--- building final phase
2021-07-10T22:39:46.1204570Z make --no-print-directory -f ghc.mk phase=final check
2021-07-10T22:39:49.0161381Z make[1]: *** No rule to make target 'check'.  Stop.
2021-07-10T22:39:49.0186932Z make: *** [Makefile:128: check] Error 2
2021-07-10T22:39:49.0191464Z => ERROR: ghc-8.8.4_3: do_check: '${make_cmd} ${make_check_args} ${make_check_target}' exited with 2
2021-07-10T22:39:49.0193972Z => ERROR:   in do_check() at common/build-style/gnu-configure.sh:33
2021-07-10T22:39:49.0462777Z ##[error]Process completed with exit code 1.

File: srcpkgs/ghc/template [hunk 1]
```diff
@@ -2,7 +2,7 @@
 pkgname=ghc
 # Keep this synchronized with http://www.stackage.org/lts
 version=8.8.4
-revision=2
+revision=3
```
Comment (ericonr) [inline srcpkgs/ghc/template:5]: If this is only to fix the build, I don't think we need to revbump.

Comment (chexum) [reply srcpkgs/ghc/template:5]: Fair enough - anyone having a previously  compiled version is not likely interested to have a new revision forced onto them.  Would I need to push an updated commit?

Comment (ericonr) [reply srcpkgs/ghc/template:5]: Yes, please.

**Comment by ericonr** [comment]
Nice find, thanks!

**Pull request merged**